server/src/main/java/org/elasticsearch/index/mapper/DynamicTemplate.java

```diff
@@ -217,41 +220,78 @@ static DynamicTemplate parse(String name, Map<String, Object> conf) throws Mappe
         }
 
         final MatchType matchType = MatchType.fromString(matchPattern);
+        List<String> allPatterns = Stream.concat(
+            Stream.concat(match.stream(), unmatch.stream()),
+            Stream.concat(pathMatch.stream(), pathUnmatch.stream())
+        ).collect(Collectors.toList());
```

Comment:
I had forgotten that `concat` only takes two arguments.  This might be better as `Stream.of(match.stream(), unmatch.stream(), pathMatch.stream(), pathUnmatch.stream()).flatMap(s -> s).toList()`